core/src/kmx/kmx_plus.cpp

```diff
@@ -0,0 +1,224 @@
+    } else {
+        printf("  \t UTF-32:U+%04X\n", entry.to);
+    }
+  }
+}
+
+static void
+dump_kmxplus_loca(const uint8_t* /*data*/, const COMP_KMXPLUS_LOCA* loca) {
+  if(loca == NULL) {
+    printf("! could not load 'loca' section\n");
+    return;
+  }
+  dump_kmxplus_header((const COMP_KMXPLUS_HEADER*)loca);
+  // TODO-LDML
+  for(KMX_DWORD i=0; i<loca->count; i++) {
+    printf(" Locale #%d: #0x%X\n", i, loca->entries[i].locale);
+  }
+}
+
+static void
+dump_kmxplus_meta(const uint8_t* /*data*/, const COMP_KMXPLUS_META* meta) {
+  if(meta == NULL) {
+    printf("! could not load 'meta' section\n");
+    return;
+  }
+  dump_kmxplus_header((const COMP_KMXPLUS_HEADER*)meta);
+  printf(" name:\t#0x%X\n", meta->name);
+  printf(" author:\t#0x%X\n", meta->author);
+  printf(" conform:\t#0x%X\n", meta->conform);
+  printf(" layout:\t#0x%X\n", meta->layout);
+  printf(" normalization:\t#0x%X\n", meta->normalization);
+  printf(" indicator:\t#0x%X\n", meta->indicator);
+  printf(" settings:\t0x%X\n", meta->settings);
+}
+
+static void
+dump_kmxplus_vkey(const uint8_t* /*data*/, const COMP_KMXPLUS_VKEY* vkey) {
+  if(vkey == NULL) {
+    printf("! could not load 'vkey' section\n");
+    return;
+  }
+  dump_kmxplus_header((const COMP_KMXPLUS_HEADER*)vkey);
+  // TODO-LDML
+}
+
+static void
+dump_kmxplus_strs(const uint8_t* /*data*/, const COMP_KMXPLUS_STRS* strs) {
+  dump_kmxplus_header((const COMP_KMXPLUS_HEADER*)strs);
+  printf("strs: count 0x%X\n", strs->count);
+  for (KMX_DWORD i=0; i<strs->count; i++) {
+    KMX_WCHAR buf[BUFSIZ];
+    printf("#0x%X: ", i);
+    PKMX_WCHAR str = strs->get(i, buf, BUFSIZ);
+    if (!str) {
+        printf("NULL/ERR\n");
+        continue;
+    }
+    for(int j=0; str[j] && j<0x30; j++) {
+        if (str[j] < 0x7F && str[j] != 0x0020 && str[j] > 0x20) {
+            putchar(str[j]);
+        } else {
+            printf("U+%04X ", str[j]);
+        }
+    }
+    printf("\n");
+  }
+}
+
+static void
+dump_kmxplus_sect(const uint8_t* data, const COMP_KMXPLUS_SECT* sect) {
+  dump_kmxplus_header((const COMP_KMXPLUS_HEADER*)sect);
+  printf("sect: total 0x%X\n", sect->total);
+  printf("sect: count 0x%X\n", sect->count);
+
+  for (KMX_DWORD i = 0; i < sect->count; i++) {
+    const COMP_KMXPLUS_SECT_ENTRY& entry = sect->entries[i];
+    dump_section_name(entry.sect);
+    printf(" sect#%d: %X @ %X\n", i, entry.sect, entry.offset);
+    const uint8_t* entrydata = (data+entry.offset);
+    switch(entry.sect) {
+        case LDML_SECTION_KEYS:
+            dump_kmxplus_keys(data, as_kmxplus_keys(entrydata));
+            break;
+        case LDML_SECTION_LOCA:
+            dump_kmxplus_loca(data, as_kmxplus_loca(entrydata));
+            break;
+        case LDML_SECTION_META:
+            dump_kmxplus_meta(data, as_kmxplus_meta(entrydata));
+            break;
+        case LDML_SECTION_SECT:
+            printf("! Cowardly refusing to dump nested 'sect' section.\n");
```

Comment:
This should be an error, shouldn't it? We don't allow nested `sect` sections?

Reply:
Right. This might become a validator but it's currently just a debug statement. 

Is there a good place for a "dump" facility? Separate tool?

Reply:
Yes, we should have a `core/src/tools` folder (per our [repo filesystem layout doc](https://docs.google.com/document/d/1gWlgDA_re33i_ny4FiBrjuZLJLBJqlIAbspRxI55FQk/edit)). A `kmxplusdump` tool would belong there pretty neatly. Or else in `developer/src/tools` but I think under `core` makes sense given the C++ codebase.

Note that `core/tests` should probably be moving to `core/src/test` according to that doc (cleanup for another day?).